gspread/spreadsheet.py

```diff
@@ -287,11 +287,11 @@ def get_worksheet_by_id(self, id):
 
         try:
             item = finditem(
-                lambda x: x["properties"]["sheetId"] == id,
+                lambda x: x["properties"]["sheetId"] == int(id),
                 sheet_data["sheets"],
             )
             return Worksheet(self, item["properties"])
-        except (StopIteration, KeyError):
+        except (StopIteration, KeyError, ValueError):
             raise WorksheetNotFound("id {} not found".format(id))
```

Comment:
This should be a separate except clause. However, `str(id)` should not throw a ValueError so could also be removed.

Reply:
Because of [above](https://github.com/burnash/gspread/pull/1269#discussion_r1288756690) I believe the `ValueError` is still necessary, the question then is leave it as one exception handler or create a separate one?  Maybe something like this could help point someone in the right direction:
```python
try:
  # ...
except ValueError:
  raise WorksheetNotFound("id {} not found, id must be of type int".format(id))
```